doc/developers/developing_scikit.rst

```diff
@@ -0,0 +1,761 @@
+We expect code coverage of new features to be at least around 90%.
+
+For guidelines on how to use ``pytest`` efficiently, see the
+:ref:`pytest_tips`.
+
+Writing matplotlib related tests
+--------------------------------
+
+Test fixtures ensure that a set of tests will be executing with the appropriate
+initialization and cleanup. The scikit-learn test suite implements a fixture
+which can be used with ``matplotlib``.
+
+``pyplot``
+    The ``pyplot`` fixture should be used when a test function is dealing with
+    ``matplotlib``. ``matplotlib`` is a soft dependency and is not required.
+    This fixture is in charge of skipping the tests if ``matplotlib`` is not
+    installed. In addition, figures created during the tests will be
+    automatically closed once the test function has been executed.
+
+To use this fixture in a test function, one needs to pass it as an
+argument::
+
+    def test_requiring_mpl_fixture(pyplot):
+        # you can now safely use matplotlib
+
+Workflow to improve test coverage
+---------------------------------
+
+To test code coverage, you need to install the `coverage
+<https://pypi.org/project/coverage/>`_ package in addition to pytest.
+
+1. Run 'make test-coverage'. The output lists for each file the line
+    numbers that are not tested.
+
+2. Find a low hanging fruit, looking at which lines are not tested,
+    write or adapt a test specifically for these lines.
+
+3. Loop.
+
+
+Issue Tracker Tags
+------------------
+All issues and pull requests on the
+`GitHub issue tracker <https://github.com/scikit-learn/scikit-learn/issues>`_
+should have (at least) one of the following tags:
+
+:Bug / Crash:
+    Something is happening that clearly shouldn't happen.
+    Wrong results as well as unexpected errors from estimators go here.
+
+:Cleanup / Enhancement:
+    Improving performance, usability, consistency.
+
+:Documentation:
+    Missing, incorrect or sub-standard documentations and examples.
+
+:New Feature:
+    Feature requests and pull requests implementing a new feature.
+
+There are four other tags to help new contributors:
+
+:good first issue:
+    This issue is ideal for a first contribution to scikit-learn. Ask for help
+    if the formulation is unclear. If you have already contributed to
+    scikit-learn, look at Easy issues instead.
+
+:Easy:
+    This issue can be tackled without much prior experience.
+
+:Moderate:
+    Might need some knowledge of machine learning or the package,
+    but is still approachable for someone new to the project.
+
+:help wanted:
+    This tag marks an issue which currently lacks a contributor or a
+    PR that needs another contributor to take over the work. These
+    issues can range in difficulty, and may not be approachable
+    for new contributors. Note that not all issues which need
+    contributors will have this tag.
+
+.. _coding-guidelines:
+
+Coding guidelines
+=================
+
+The following are some guidelines on how new code should be written. Of
```

Comment:
Thesl issue suggests that the top matter from this section goes into the contributing guide, not here. Or maybe we can just say that the following are some standards used within scikit-learn, which may be appropriate to adopt in external projects for the sake of consistency

Reply:
@jnothman Thanks for the feedback. I have made all the requested changes. But I am still not sure about how to properly handle the 'Coding guidelines' section in the new file. I just changed the first line to reflect what you had commented above. But if further modifications are needed please let me know.